### ISSUE TITLE
S3 proxying isn't working out of the box, there is a bugfix on verdaccio only repo??

### ISSUE THREAD

**Issue by jraubswitch** [Remitly/verdaccio-s3-storage#26]
Through much searching and debugging I have s3 working with proxying now. I had to install the fix version `npm install verdaccio-s3-storage@0.1.0-fix --registry https://registry.verdaccio.org` from the registry.verdaccio.org which I found from the example dockerfile. Can this be put into the main code please? it looks like the only pertinent piece is ignoring 404s from s3. 
https://registry.verdaccio.org/-/web/detail/verdaccio-s3-storage

https://github.com/Remitly/verdaccio-s3-storage/blob/master/src/s3PackageManager.js#L243

https://github.com/verdaccio/docker-examples/blob/master/amazon-s3-docker-example/s3Plugin/Dockerfile#L3

./lib/s3PackageManager.js <--- the 0.1.0-fix version file that was copied to make it work.

```
--- ./lib/s3PackageManager.js
+++ /usr/local/app/node_modules/verdaccio-s3-storage/lib/s3PackageManager.js
@@ -231,7 +233,7 @@
       // verdaccio force garbage collects a stream on 404, so we can't emit more
       // than one error or it'll fail
       // https://github.com/verdaccio/verdaccio/blob/c1bc261/src/lib/storage.js#L178
-      if (statusCode !== 404) {
+
         if (headers['content-length']) {
           const contentLength = parseInt(headers['content-length'], 10);

@@ -246,7 +248,6 @@
           readTarballStream.emit('content-length', contentLength);
           // we know there's content, so open the stream
           readTarballStream.emit('open');
-        }
       }
     }).createReadStream();
```

**Comment by jraubswitch** [comment]
@juanpicado

The actual applied diff that I used

```
diff --git a/src/s3PackageManager.js b/src/s3PackageManager.js
index af91a93..64bfe7a 100644
--- a/src/s3PackageManager.js
+++ b/src/s3PackageManager.js
@@ -239,8 +239,7 @@ export default class S3PackageManager implements ILocalPackageManager {
         // verdaccio force garbage collects a stream on 404, so we can't emit more
         // than one error or it'll fail
         // https://github.com/verdaccio/verdaccio/blob/c1bc261/src/lib/storage.js#L178
-
-        if (headers['content-length']) {
+        if (statusCode !== 404 && headers['content-length']) {
           const contentLength = parseInt(headers['content-length'], 10);

           // not sure this is necessary
```

**Comment by snetty** [comment]
0.1.0-fix version doesn't appear to exist anymore.. we can't currently use verdaccio because s3 proxying is failing

**Comment by apexskier** [comment]
@snetty I'm not sure who actually published the 0.1.0-fix version, and it looks like it was published to a staging location?

If you submit a PR I can publish an official release.

**Comment by snetty** [comment]
I'll give it a go, am I right in thinking that the tests need to connect to s3?

**Comment by apexskier** [comment]
Yeah, I think when I initially built this I tested against a test bucket. I relied on my environment aws creds and the tests use the env variable [VERDACCIO_TEST_BUCKET](https://github.com/Remitly/verdaccio-s3-storage/blob/master/src/___tests___/s3PackageManager.test.js#L19)

**Comment by soulchild** [comment]
This issue is currently keeping us from using Verdaccio at all. Any chance there's gonna be a public release incorporating this fix soon(ish)? Thanks a lot!

**Comment by apexskier** [comment]
released as v0.2.1
